src/functional/ffront/past_passes/type_deduction.py

```diff
@@ -100,6 +97,27 @@ def visit_Call(self, node: past.Call, **kwargs):
             location=node.location,
         )
 
+    def _check_out_and_field_domain_values(self, new_func: past.Name, new_kwargs: dict):
+        if not isinstance(new_func.type, (ct.FieldOperatorType, ct.ScanOperatorType)):
+            raise GTTypeError(
+                f"Only calls `FieldOperator`s and `ScanOperators` "
+                f"allowed in `Program`, but got `{new_func.type}`."
+            )
+
+        if "out" not in new_kwargs:
+            raise GTTypeError("Missing required keyword argument(s) `out`.")
+        elif "field_domain" in new_kwargs:
+            domain_kwarg = new_kwargs["field_domain"]
+            if not isinstance(domain_kwarg, past.Dict):
+                raise GTTypeError(
+                    f"Only calls Dictionary allowed in field_domain, but got `{type(domain_kwarg)}`."
+                )
+            for domain_values in domain_kwarg.values_:
```

Comment:
Also check that keys are `Dimensions` (see also my request for changing `str` -> `Dimension` in the test).

Reply:
Before adding more checks, I propose to implement some type deduction test that you expect to fail, e.g.

`domain={1:(2,3)}`, `domain={Dim:(2,3,4)}` etc...